Secure URLs

vszakats opened the pull request:
Ref: https://github.com/Homebrew/homebrew-cask/pull/52865

Stats:
```
              before          after
   Casks:       4041      →    4041
   HTTPS:       8291      →    9254
Insecure:       2382      →    1423
    HTTP:       2374      →    1420
     FTP:          8      →       3
   Ratio:         77.68 % →      86.67 %
```

vszakats commented:
Some of changes are meant to fix wrongly configured redirects in _existing_ URLs, so in maybe 2 occasions I pointed to the English homepage. Otherwise it'd redirect to an HTTP page. Is that what you mean?

There was another single URL which now redirects to a different domain, but it's also wrongly configured to use HTTP, so I followed the redirect an upgraded to HTTPS manually.

Other than these and maybe a few similar occasions there is not "URL update" in this PR.

vszakats commented:
I can split this to any number of `.diff`s, the practical limit is the manual labour needed to create/update/manage each PR separately. Splitting this to 3-5 pieces looks fine, but above that it'd save a lot of unnecessary time/clicking if I'd just attach a list of `.diff` files that you can then apply directly, if that can work with you.

vszakats commented:
Replaced that particular redirect (`aluxian-messenger`) with an `archive.org` link. The product is deprecated.
Also went through all broken links and fixed/replaced a bunch of them.

vszakats commented:
I had run the redir checker with this patch applied, over `url` and `appcast` stanzas (taking out known good GitHub and SourceForge URLs). After applying fixes for some of them, and adding a fully secure (but not officially linked to) link as comment for some others, the number of remaining links that land on insecure HTTP is 65 (out of 3545).

<details><summary>Click for list</summary><p>
```
https://assets.unigine.com/d/Unigine_Heaven-4.0.dmg -> http://m11-assets.unigine.com/d/Unigine_Heaven-4.0.dmg
https://assets.unigine.com/d/Unigine_Valley-1.0.dmg -> http://m12-assets.unigine.com/d/Unigine_Valley-1.0.dmg
https://binaries.openttd.org/releases/1.8.0/openttd-1.8.0-macosx-universal.zip -> http://cz.binaries.openttd.org/binaries/releases/1.8.0/openttd-1.8.0-macosx-universal.zip
https://download.kde.org/stable/digikam/digiKam-5.9.0-01-MacOS-x86-64.pkg -> http://ftp.kaist.ac.kr/kde/stable/digikam/digiKam-5.9.0-01-MacOS-x86-64.pkg
https://download.kde.org/stable/krita/4.1.3/krita-4.1.3.dmg -> http://kde.mirror.anlx.net/stable/krita/4.1.3/krita-4.1.3.dmg
https://download.qianqian.com/channel/2/web_daohang -> http://online.cdn.qianqian.com/qianqian/info/3b470433bc70d25c0199ada4f74fb059.dmg
https://download.qt.io/official_releases/qtcreator/4.7/4.7.1/qt-creator-opensource-mac-x86_64-4.7.1.dmg -> http://ftp.fau.de/qtproject/official_releases/qtcreator/4.7/4.7.1/qt-creator-opensource-mac-x86_64-4.7.1.dmg
https://fetchsoftworks.com/fetch/download/Fetch_5.7.7.dmg?direct=1 -> http://fetchsoftworks.com/fetch/download/Fetch_5.7.7.dmg?direct=1
https://files.kde.org/marble/downloads/MacOSX/Marble-2.2.0.pkg -> http://ftp.icm.edu.pl/packages/kde-applicationdata/marble/downloads/MacOSX/Marble-2.2.0.pkg
https://firealpaca.com/download/mac -> http://firealpaca.com/
https://fliqlo.com/download/fliqlo_171.dmg -> http://fliqlo.com/
https://futureaudioworkshop.com/downloads/Circle-2.1.0-RC3-setup.dmg -> http://futureaudioworkshop.com/downloads/Circle-2.1.0-RC3-setup.dmg
https://futureaudioworkshop.com/downloads/extras/Circle-Wavetable-Generator.dmg -> http://futureaudioworkshop.com/downloads/extras/Circle-Wavetable-Generator.dmg
https://mbdapp.iqiyi.com/j/ot/iQIYIMedia_005.dmg -> http://cdn.data.video.iqiyi.com/cdn/pcclient/20180830/19/02/iQIYIMedia_005.dmg
https://mblock.makeblock.com/mBlock_mac_V3.4.11.zip -> http://mblockapp.oss-cn-hongkong.aliyuncs.com/mBlock_mac_V3.4.11.zip
https://sparkler.herokuapp.com/apps/3/versions/132/download -> http://s3.amazonaws.com/sparkler_versions/versions/uploads/000/000/132/original/Anvil.zip?1513340725
https://srf.baidu.com/?c=j&e=d&platform=mac -> http://imeres.baidu.com/mac/baiduinput_mac_v5.2.0.22_1000e.dmg
https://stemsmusic.wpengine.com/download/stems-creator-installer-mac/ -> http://stemsmusic.wpengine.com/download/stems-creator-installer-mac/
https://support.apple.com/downloads/DL1433/en_US/RecoveryDiskAssistant.dmg -> http://download.info.apple.com/Mac_OS_X/041-2001.20110808.QFC1d/RecoveryDiskAssistant.dmg
https://wireframesketcher.com/downloads/WireframeSketcher-latest-macosx.zip -> http://cdn.wireframesketcher.com/studio/dist/WireframeSketcher-5.1.0-macosx.zip
https://www.5kplayer.com/download/5kplayer.dmg -> http://dl1.5kplayer.com/download/5kplayer.dmg
https://www.5kplayer.com/upgrade/mac/updatecast.xml -> http://www.5kplayer.com/upgrade/mac/updatecast.xml
https://www.bannister.org/cgi-bin/download.cgi?nestopia -> http://maczone.free.fr/DIRECT_LINKING_NOT_PERMITTED_124398623984/nestopia-142.zip
https://www.cocos2d-x.org/filedown/cocos-code-ide-mac64-1.2.0.dmg -> http://www.cocos2d-x.org/filedown/cocos-code-ide-mac64-1.2.0.dmg
https://www.deezer.com/desktop/download/artifact/darwin/x64/0.14.4 -> http://e-cdn-content.deezer.com/builds/deezer-desktop/8cF2rAuKxLcU1oMDmCYm8Uiqe19Ql0HTySLssdzLkQ9ZWHuDTp2JBtQOvdrFzWPA/darwin/x64/0.14.4/DeezerDesktop_0.14.4.dmg
https://www.digitalrebellion.com/download/cineplay?version=1440 -> http://download.digitalrebellion.com/cineplay/CinePlay_1440.dmg
https://www.digitalrebellion.com/download/fcsremover?version=3150 -> http://download.digitalrebellion.com/fcsremover/FCS_Remover_3150.dmg
https://www.digitalrebellion.com/download/kollabfolderwatcher?version=1320 -> http://download.digitalrebellion.com/kollabfolderwatcher/Kollaborate_Folder_Watcher_1320.dmg
https://www.digitalrebellion.com/download/kollabplugins?version=1120 -> http://download.digitalrebellion.com/kollabplugins/Kollaborate_Plugin_Pack_1120.dmg
https://www.digitalrebellion.com/download/kollabtransfer?version=1520 -> http://download.digitalrebellion.com/kollabtransfer/Kollaborate_Transfer_1520.dmg
https://www.digitalrebellion.com/download/markerimport?version=2180 -> http://download.digitalrebellion.com/markerimport/Marker_Import_for_Mac_2180.dmg
https://www.digitalrebellion.com/download/posthaste?version=2640 -> http://download.digitalrebellion.com/posthaste/Post_Haste_for_Mac_2640.dmg
https://www.digitalrebellion.com/download/prefman?version=4420 -> http://download.digitalrebellion.com/prefman/Preference_Manager_4420.dmg
https://www.dssw.co.uk/authbuddy/dsswauthbuddy.dmg -> http://www.dssw.co.uk/authbuddy/dsswauthbuddy-110.dmg
https://www.dssw.co.uk/sleepmonitor/dsswsleepmonitor.dmg -> http://www.dssw.co.uk/sleepmonitor/dsswsleepmonitor-302.dmg
https://www.eclipse.org/downloads/download.php?file=/modeling/mdt/papyrus/rcp/oxygen/3.0.0/papyrus-oxygen-3.0.0-macosx64.tar.gz&r=1 -> http://download.eclipse.org/modeling/mdt/papyrus/rcp/oxygen/3.0.0/papyrus-oxygen-3.0.0-macosx64.tar.gz
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/2018-09/R/eclipse-jee-2018-09-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/2018-09/R/eclipse-jee-2018-09-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/oxygen/3a/eclipse-dsl-oxygen-3a-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/oxygen/3a/eclipse-dsl-oxygen-3a-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/oxygen/3a/eclipse-javascript-oxygen-3a-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/oxygen/3a/eclipse-javascript-oxygen-3a-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/oxygen/3a/eclipse-modeling-oxygen-3a-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/oxygen/3a/eclipse-modeling-oxygen-3a-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/oxygen/3a/eclipse-parallel-oxygen-3a-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/oxygen/3a/eclipse-parallel-oxygen-3a-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/oxygen/3a/eclipse-php-oxygen-3a-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/oxygen/3a/eclipse-php-oxygen-3a-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/oxygen/3a/eclipse-rcp-oxygen-3a-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/oxygen/3a/eclipse-rcp-oxygen-3a-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/oxygen/3a/eclipse-scout-oxygen-3a-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/oxygen/3a/eclipse-scout-oxygen-3a-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/oxygen/3a/eclipse-testing-oxygen-3a-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/oxygen/3a/eclipse-testing-oxygen-3a-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/photon/R/eclipse-committers-photon-R-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/photon/R/eclipse-committers-photon-R-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/photon/R/eclipse-cpp-photon-R-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/photon/R/eclipse-cpp-photon-R-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/photon/R/eclipse-java-photon-R-macosx-cocoa-x86_64.dmg&r=1 -> http://download.eclipse.org/technology/epp/downloads/release/photon/R/eclipse-java-photon-R-macosx-cocoa-x86_64.dmg
https://www.eclipse.org/downloads/download.php?r=1&file=/mat/1.8/rcp/MemoryAnalyzer-1.8.0.20180604-macosx.cocoa.x86_64.zip -> http://download.eclipse.org/mat/1.8/rcp/MemoryAnalyzer-1.8.0.20180604-macosx.cocoa.x86_64.zip
https://www.fadingred.com/icolors/updates.xml -> http://www.fadingred.com/icolors/updates.xml
https://www.fenrir-inc.com/services/download.php?file=Sleipnir.dmg -> http://download2.fenrir.co.jp/Sleipnir.dmg?fid=0&lid=1041&ext=Sleipnir.dmg
https://www.jrtstudio.com/files/iSyncr%20Desktop%205.14.0.pkg -> http://www.jrtstudio.com/files/iSyncr%20Desktop%205.14.0.pkg
https://www.karelia.com/files/13/Tangerine!.zip -> http://distrib.karelia.com/downloads/Tangerine!-4011.zip
https://www.karelia.com/files/8/Sandvox.dmg -> http://distrib.karelia.com/downloads/_Sandvox-29586.zip
https://www.livesurface.com/getcontextfile.php?du=b64c31a5-e26a-4a09-9f01-dd362e376c89&d=LiveSurface_Context_1.1.718.dmg -> http://1590386904.rsc.cdn77.org/94775618/LiveSurface_Context_1.1.718.dmg
https://www.macupdate.com/download/34101/SafariSort.dmg -> http://dl.macupdate.com/prod/SafariSort.dmg
https://www.shinywhitebox.com/store/file.php?p=1 -> http://www.shinywhitebox.com/downloads/iShowU_1.94.5.dmg
https://www.shinywhitebox.com/store/file.php?p=12 -> http://www.shinywhitebox.com/downloads/iShowU_Studio_1.7.6.dmg
https://www.shinywhitebox.com/store/file.php?p=14 -> http://www.shinywhitebox.com/downloads/instant/iShowU_Instant_1.2.8.dmg
https://www.shinywhitebox.com/store/file.php?p=5 -> http://www.shinywhitebox.com/downloads/iShowU_HD_2.3.21.dmg
https://www.syniumsoftware.com/download/logoist.dmg -> http://download.syniumsoftware.com/Logoist/Logoist%203.2%20Demo.dmg
https://www.xmind.net/xmind/downloads/xmind-8-update8-macosx.dmg -> http://dl2.xmind.net/xmind-downloads/xmind-8-update8-macosx.dmg
https://xritephoto.com/downloader.aspx?FileID=1809&Type=M&returnurl=%2fph_product_overview.aspx%3fID%3d1115%26Action%3dSupport%26SoftwareID%3d1809%26catid%3d149 -> http://downloads.xrite.com/downloads/software/ColorMunki/Photo/v1.2.4/mac/ColorMunkiPhoto.zip
```
</p></details>

vszakats commented:
Let me know if you have any questions/concerns regarding any of the updates, and on how to split it to separate commits/PRs.

vitorgalvao commented:
> Replaced that particular (`aluxian-messenger`) homepage redirect with an `archive.org` link. The app is deprecated.

Remove the cask instead.

commitay commented:
Three PRs would be fine. I've moved the docs commit to https://github.com/Homebrew/homebrew-cask/pull/53025.

These audit and style issues need to be addressed, as it's mostly SourceForge related it might be easier to move those casks to their own PR.
```
audit for letterfix: warning
 - OSDN URL format incorrect. See https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#sourceforgeosdn-urls
audit for miditrail: warning
 - OSDN URL format incorrect. See https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#sourceforgeosdn-urls
audit for nndd: warning
 - OSDN URL format incorrect. See https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#sourceforgeosdn-urls
```
```
== aria-maestosa.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== arrsync.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== audioslicer.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== blobby-volley2.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== brain-workshop.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== bricksmith.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== brushviewql.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: brushviewer.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== burn.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== cactus.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: update.cactusformac.com does not match archive.org, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== camed.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== chatology.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: d60ism0l33mmr.cloudfront.net does not match flexibits.com, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== chessx.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== coccinellida.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== cord.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== cuda-z.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== davmail.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== djv.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== djview.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== double-commander.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== doublecommand.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: doublecommand.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== emin-webpquicklook.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: raw.githubusercontent.com does not match github.com, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== excalibur.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: excalibur.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== fredm-fuse.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== freelib.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== freemind.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== freeplane.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== fugenschnitzer.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== gawker.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== gifs.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: raw.githubusercontent.com does not match github.com, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== golly.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== grandperspective.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== gretl.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== gtkwave.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== handy-outliner.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== hugin.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== inc.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: inc-static.kippt.com.s3.amazonaws.com does not match s3.amazonaws.com/inc-static.kippt.com/apps/inc-osx.zip
== iupx.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== jumpcut.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== kdiff3.rb ==
C:  6:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== kid3.rb ==
C:  6:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== klatexformula.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== latexdraw.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== leonardo.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: marginsoftware.de does not match margin.de, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== letterfix.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: osdn.net does not match osdn.jp, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== luminance-hdr.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== lynkeos.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== m64py.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== macbiff.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== macdjview.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== mame.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== media-converter.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== meteorologist.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== mongobooster.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: s3.mongobooster.com does not match nosqlbooster.com, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== nndd.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: osdn.net does not match osdn.jp, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== paintbrush.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== panconvert.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== plotdigitizer.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== posterazor.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== privateer-gold.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== qtspim.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== quakespasm.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== scidavis.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== scidvsmac.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== scrabble3d.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== seashore.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== skim.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== speeds-at-ver.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: raw.githubusercontent.com does not match github.com, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== sqlexplorer.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== the-cheat.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: chazmcgarvey.github.io does not match github.com, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== the-ur-quan-masters.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== tuntap.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== ultrastardeluxe.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== upm.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
== vmpk.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: #  downloads.sourceforge.net/vmpk was verified as official when first introduced to the cask does not match downloads.sourceforge.net/vmpk/vmpk-
== webarchiveextractor.rb ==
C:  5:  3: Cask/HomepageMatchesUrl: downloads.sourceforge.net does not match sourceforge.io, a comment has to be added above the url stanza. For details, see https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment
```

vszakats commented:
As for OSDN URLs, I believe it'd be better to update the documentation to match the URLs currently advertised on OSDN download pages (there has been some overhaul done here lately). These new URLs (and also the ones I used — to stay in sync with `homebrew-core`) do support HTTPS URLs. (Though sadly there is no guarantee that the mirror logic picks an HTTPS mirror.)

Here's the official short form:
https://osdn.net/dl/letter-fix/LetterFix-2.7.0.1.dmg

See: https://osdn.net/projects/letter-fix/releases/

I'll try to correct those style issues in the meantime.

commitay commented:
Deleted `aluxian-messenger` in https://github.com/Homebrew/homebrew-cask/pull/53026

vszakats commented:
One more thought: The officially blessed SourceForge user pages are `https://<project>.sourceforge.io/` or `https://sourceforge.net/projects/<project>/` while matching downloads are typically under `https://downloads.sourceforge.net/`, so it may useful to assume this logic by default. Likewise `raw.githubusercontent.com` would safely match `github.com` and `<project>.github.io`.

commitay commented:
Fixed `r-app` permissions in https://github.com/Homebrew/homebrew-cask/pull/53027

vitorgalvao commented:
@vszakats Regarding your last comment, I’d rather keep the rules as they are. They took a lot of work and iteration to get to this point, and we considered/tested those cases, such as not needing a comment on github.io/github.com cases, and reached the conclusion it’s safer this way, and keeps us on our toes.

vszakats commented:
Hopefully fixed all warnings. (Except the osdn.net `url` style, until further conclusion.)

vszakats commented:
Okay, I've already fixed those — but, I'd like to insist on rethinking the OSDN download rules. OSDN (JP) went through a lot of changes in the last few years, and the current style checker/docs reflect the state that was true 2 years ago and before that, but not anymore. The new URL-style is much simpler and cleaned up most of the former mess. You can confirm the new one by looking up any OSDN JP project page, and check how the download URLs look there:
![screenshot 2018-10-07 at 01 28 07](https://user-images.githubusercontent.com/1446897/46576679-6df5b580-c9d0-11e8-9a7f-3867f711d219.png)

There is no `.jp`, there is HTTPS, there is mirroring behind, and no duplicate URL parts, nor any magic numbers like `release_id`. Can't see any reason not to use the official ones for this particular site, especially if they are superior than the obsolete ones.

vszakats commented:
Undone the docs, exec attr and .rb deletion hunks/commit.

commitay commented:
As I said above, can you split this up into a couple of PRs please, we're picky about URLs and trying to review this many casks/changes at once isn't easy with the style and audit issues.

vitorgalvao commented:
> I'd like to insist on rethinking the OSDN download rules

Yes, completely open to that. For context, the reason we enforce a a style for those URLs is that they invariably redirect to a regional CDN, and users kept submitting those. By enforcing a style, we ensure we get the “neutral” URL.

My comment was solely referring to the relationship of `homepage` with `url`.

commitay commented:
`# downloads.sourceforge.net was verified` 

should be 

`# downloads.sourceforge.net/{project} was verified`

vszakats commented:
Brew and Homebrew-Cask patches to accept new-style OSDN URLs: (lightly tested)
```diff
diff --git a/Library/Homebrew/cask/audit.rb b/Library/Homebrew/cask/audit.rb
index 8a5d0b53d..aa33a608d 100644
--- a/Library/Homebrew/cask/audit.rb
+++ b/Library/Homebrew/cask/audit.rb
@@ -290,7 +290,7 @@ module Cask
     end
 
     def bad_osdn_url?
-      bad_url_format?(/osd/, [%r{\Ahttps?://([^/]+.)?dl\.osdn\.jp/}])
+      bad_url_format?(/osd/, [%r{\Ahttps://osdn\.net/}])
     end
 
     def check_generic_artifacts
diff --git a/Library/Homebrew/test/support/fixtures/cask/Casks/osdn-correct-url-format.rb b/Library/Homebrew/test/support/fixtures/cask/Casks/osdn-correct-url-format.rb
index 2a16272bd..49b7d4744 100644
--- a/Library/Homebrew/test/support/fixtures/cask/Casks/osdn-correct-url-format.rb
+++ b/Library/Homebrew/test/support/fixtures/cask/Casks/osdn-correct-url-format.rb
@@ -1,6 +1,6 @@
 cask 'osdn-correct-url-format' do
   version '1.2.3'
 
-  url 'https://user.dl.osdn.jp/something/id/Something-1.2.3.dmg'
-  homepage 'https://osdn.jp/projects/something/'
+  url 'https://osdn.net/dl/something/Something-1.2.3.dmg'
+  homepage 'https://osdn.net/projects/something/'
 end
```

```diff
diff --git a/Library/Homebrew/cask/audit.rb b/Library/Homebrew/cask/audit.rb
index 8a5d0b53d..aa33a608d 100644
--- a/Library/Homebrew/cask/audit.rb
+++ b/Library/Homebrew/cask/audit.rb
@@ -290,7 +290,7 @@ module Cask
     end
 
     def bad_osdn_url?
-      bad_url_format?(/osd/, [%r{\Ahttps?://([^/]+.)?dl\.osdn\.jp/}])
+      bad_url_format?(/osd/, [%r{\Ahttps://osdn\.net/}])
     end
 
     def check_generic_artifacts
diff --git a/Library/Homebrew/test/support/fixtures/cask/Casks/osdn-correct-url-format.rb b/Library/Homebrew/test/support/fixtures/cask/Casks/osdn-correct-url-format.rb
index 2a16272bd..49b7d4744 100644
--- a/Library/Homebrew/test/support/fixtures/cask/Casks/osdn-correct-url-format.rb
+++ b/Library/Homebrew/test/support/fixtures/cask/Casks/osdn-correct-url-format.rb
@@ -1,6 +1,6 @@
 cask 'osdn-correct-url-format' do
   version '1.2.3'
 
-  url 'https://user.dl.osdn.jp/something/id/Something-1.2.3.dmg'
-  homepage 'https://osdn.jp/projects/something/'
+  url 'https://osdn.net/dl/something/Something-1.2.3.dmg'
+  homepage 'https://osdn.net/projects/something/'
 end
```

vszakats commented:
@commitay: Sigh, ok, I have just remove all the subdirectories from those comments, after realising that the checker doesn't care for it :(

vszakats commented:
It'd be very unpractical to split this PR into multiple ones, till the patch is in flux. Lot's of busywork anyway.

commitay commented:
> I'm ready to take this exact patch and submit it as three PRs

Please do.

commitay commented:
These casks in the fonts repo need to be updated to comply with the audit changes.


- font-genjyuugothic
- font-genjyuugothic-l
- font-genjyuugothic-x
- font-genshingothic
- font-hanamina
- font-koruri
- font-m-plus
- font-migmix-1m
- font-migmix-1p
- font-migmix-2m
- font-migmix-2p
- font-migu-1c
- font-migu-1m
- font-migu-1p
- font-migu-2m
- font-ocr
- font-rounded-m-plus

vszakats commented:
Added the same patch split in three equal PRs, (auto-)referenced above.

commitay commented:
Thanks, closing this in favour of the open PRs.
